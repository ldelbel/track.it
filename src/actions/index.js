const changePage = page => ({
  type: 'change_page',
  payload: page,
});

export default changePage;