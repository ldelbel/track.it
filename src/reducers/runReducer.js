import { ADD_RUNNING_SESSION, SET_USER, FILL_LIST } from '../actions';

let initialState = {
  id: 0,
  username: '',
  runningSessions: [],
};

if (localStorage.user) {
  initialState = {
    id: localStorage.id,
    username: localStorage.user,
    runningSessions: JSON.parse(localStorage.runningSessions),
  };
}

const runReducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_RUNNING_SESSION:
      return {
        ...state,
        runningSessions: state.runningSessions.concat(action.payload),
      };
    case SET_USER:
      return {
        ...state,
        username: action.payload,
      };
    case FILL_LIST:
      return {
        ...state,
        id: action.payload[0],
        runningSessions: action.payload[1],
      };
    default:
      return state;
  }
};

export default runReducer;
