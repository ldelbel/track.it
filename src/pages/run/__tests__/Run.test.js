import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import Run from '../index';
import runReducer from '../../../reducers/runReducer';
import { addRunningSession } from '../../../actions';

describe('run page',() => {
    it('renders correctly', () => {
        const store = createStore(runReducer);
        const run = renderer.create(
          <Provider store={store}>
            <Run addRunningSession={addRunningSession} id={1} />
          </Provider>,
        ).toJSON();
        expect(run).toMatchSnapshot();
    });
})