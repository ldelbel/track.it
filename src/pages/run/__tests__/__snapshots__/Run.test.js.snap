// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`run page renders correctly 1`] = `
<main
  className="container"
>
  <div
    className="content"
  >
    <div
      className="arrow"
    >
      <button
        aria-label="back"
        onClick={[Function]}
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M7.707 14.707a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 1.414L5.414 9H17a1 1 0 110 2H5.414l2.293 2.293a1 1 0 010 1.414z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
    <span
      className="content__title"
    >
      SET YOUR GOAL FOR TODAY
    </span>
    <div
      className="content__inputs"
    >
      <div
        className="input"
      >
        <button
          className="input__btn_add"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <div
          className="input__display"
        >
          0
        </div>
        <button
          className="input__btn_sub"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="input"
      >
        <button
          className="input__btn_add"
          data-testid="plusOneKm"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <div
          className="input__display"
        >
          0
        </div>
        <button
          className="input__btn_sub"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="dot"
      >
        <p>
          .
        </p>
      </div>
      <div
        className="input"
      >
        <button
          className="input__btn_add"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <div
          className="input__display"
        >
          0
        </div>
        <button
          className="input__btn_sub"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="input"
      >
        <button
          className="input__btn_add"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <div
          className="input__display"
        >
          0
        </div>
        <button
          className="input__btn_sub"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="measure"
      >
        Km
      </div>
    </div>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      Start Running Session
    </button>
  </div>
</main>
`;
